src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,32 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public int hashCode() {
+        if (width != null && length != null && color != null) {
```

Comment:
each check should be separate, cause you can have situation, where only one field is null and other are ok, so you want to take them into account